feature/experimental/gribi/otg_tests/encap_decap_scale/encap_decap_scale_test.go

```diff
@@ -0,0 +1,1397 @@
+	"github.com/openconfig/featureprofiles/internal/attrs"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/gribi"
+	"github.com/openconfig/gribigo/chk"
+	"github.com/openconfig/gribigo/constants"
+	"github.com/openconfig/gribigo/fluent"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Settings for configuring the baseline testbed with the test
+// topology.
+//
+// The testbed consists of ate:port1 -> dut:port1
+// and dut:port2 -> ate:port2.
+// There are DefaultVRFIPv4NHCount SubInterfaces between dut:port2
+// and ate:port2
+//
+//   - ate:port1 -> dut:port1 subnet 192.0.2.0/30
+//   - ate:port2 -> dut:port2 DefaultVRFIPv4NHCount Sub interfaces, e.g.:
+//   - ate:port2.0 -> dut:port2.0 VLAN-ID: 0 subnet 198.18.192.0/30
+//   - ate:port2.1 -> dut:port2.1 VLAN-ID: 1 subnet 198.18.192.4/30
+//   - ate:port2.2 -> dut:port2.2 VLAN-ID: 2 subnet 198.18.192.8/30
+//   - ate:port2.i -> dut:port2.i VLAN-ID i subnet 198.18.x.(4*i)/30 (out of subnet 198.18.192.0/18)
+const (
+	ipv4PrefixLen           = 30 // ipv4PrefixLen is the ATE and DUT interface IP prefix length
+	vrf1                    = "VRF-A"
+	vrf2                    = "VRF-B"
+	vrf3                    = "VRF-C"
+	IPBlockDefaultVRF       = "198.18.128.0/18"
+	IPBlockNonDefaultVRF    = "198.18.0.0/17"
+	tunnelSrcIP             = "198.51.100.99" // tunnelSrcIP represents Source IP of IPinIP Tunnel
+	policyName              = "redirect-to-VRF1"
+	StaticMAC               = "00:1A:11:00:00:01"
+	subifBaseIP             = "198.18.192.0"
+	nextHopStartIndex       = 101 // set > 2 to avoid overlap with backup NH ids 1&2
+	nextHopGroupStartIndex  = 101 // set > 2 to avoid overlap with backup NHG ids 1&2
+	dscpEncapA1             = 10
+	dscpEncapA2             = 18
+	dscpEncapB1             = 20
+	dscpEncapB2             = 28
+	dscpEncapC1             = 30
+	dscpEncapC2             = 38
+	dscpEncapD1             = 40
+	dscpEncapD2             = 48
+	dscpEncapNoMatch        = 50
+	ipv4OuterSrc111WithMask = "198.51.100.111/32"
+	ipv4OuterSrc222WithMask = "198.51.100.222/32"
+	ipv4OuterSrc222         = "198.51.100.222"
+	magicMac                = "02:00:00:00:00:01"
+	prot4                   = 4
+	prot41                  = 41
+	vrfPolW                 = "vrf_selection_policy_w"
+	niDecapTeVrf            = "DECAP_TE_VRF"
+	niEncapTeVrfA           = "ENCAP_TE_VRF_A"
+	niEncapTeVrfB           = "ENCAP_TE_VRF_B"
+	niEncapTeVrfC           = "ENCAP_TE_VRF_C"
+	niEncapTeVrfD           = "ENCAP_TE_VRF_D"
+	niTeVrf111              = "TE_VRF_111"
+	niTeVrf222              = "TE_VRF_222"
+	niDefault               = "DEFAULT"
+	IPBlockEncapA           = "20.1.64.1/15"  // IPBlockEncapA represents the ipv4 entries in EncapVRFA
```

Comment:
Should we use different IP address prefixes, such as private addresses (RFC 1918) or the TEST-NET blocks reserved in RFC 5735? The prefixes defined in this code are registered to existing users on the Internet (e.g. Microsoft) and it would be safer to use ranges that would be captured by bogon filters, just in case...

Reply:
AS Per contributing.md, we only have three ip subnets.  Out of these we already have used 2 subnets. One for interface ip addressing and other one for VRF-A/B/C.
Remaining address space is /24 which does not provide addresses which are needed for scaling.
Hence we are using other ip blocks.